spec/concepts/school_student/list_spec.rb

```diff
@@ -5,49 +5,99 @@
+      response = described_class.call(school:, token:)
+      expect(response.success?).to be(true)
+    end
 
-  it 'makes a profile API call' do
-    described_class.call(school:, token:)
+    it 'makes a profile API call' do
+      described_class.call(school:, token:)
 
-    # TODO: Replace with WebMock assertion once the profile API has been built.
-    expect(ProfileApiClient).to have_received(:list_school_students).with(token:, school_id: school.id, student_ids: [student.id])
+      # TODO: Replace with WebMock assertion once the profile API has been built.
+      expect(ProfileApiClient).to have_received(:list_school_students).with(token:, school_id: school.id, student_ids: students.map(&:id))
+    end
+
+    it 'returns a school students JSON array' do
+      response = described_class.call(school:, token:)
+      expect(response[:school_students].size).to eq(3)
+    end
+
+    it 'returns the school students in the operation response' do
+      response = described_class.call(school:, token:)
+      students.each do |student|
+        expected_user = User.new(id: student.id, name: student.name, username: student.username)
+        expect(response[:school_students]).to include(expected_user)
+      end
+    end
   end
 
-  it 'returns the school students in the operation response' do
-    response = described_class.call(school:, token:)
-    expected_user = User.new(id: student.id, name: 'name', username: 'username')
-    expect(response[:school_students].first).to eq(expected_user)
+  context 'with student_ids' do
+    let(:student_ids) { students.map(&:id).take(2) }
+    let(:filtered_students) { students.select { |student| student_ids.include?(student.id) } }
+
+    before do
+      student_attributes = filtered_students.map do |student|
+        { id: student.id, name: student.name, username: student.username }
+      end
+      stub_profile_api_list_school_students(school:, student_attributes:)
+    end
+
+    it 'returns a successful operation response' do
+      response = described_class.call(school:, token:, student_ids:)
+      expect(response.success?).to be(true)
+    end
+
+    it 'makes a profile API call' do
+      described_class.call(school:, token:, student_ids:)
+
+      # TODO: Replace with WebMock assertion once the profile API has been built.
+      expect(ProfileApiClient).to have_received(:list_school_students).with(token:, school_id: school.id, student_ids:)
+    end
+
+    it 'returns a filtered school students JSON array' do
+      response = described_class.call(school:, token:, student_ids:)
+      expect(response[:school_students].size).to eq(2)
+    end
+
+    it 'returns the filtered school students in the operation response' do
+      response = described_class.call(school:, token:, student_ids:)
+      filtered_students.each do |student|
+        expected_user = User.new(id: student.id, name: student.name, username: student.username)
+        expect(response[:school_students]).to include(expected_user)
```

Comment:
I'm a bit confused about why we're expecting a User object to be returned in the array here? More for my own understanding than anything 🤔

Reply:
Never mind, I spotted the answer further down 😅 👍